src/components/TodoField/TodoField.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ErrorType } from '../ErrorMessage/ErrorMessage';
+
+type Props = {
+  newToField: React.RefObject<HTMLInputElement>;
+  todos: Todo[];
+  onAdd: (newTodoData: string) => void;
+  todoName: string;
+  setNewTodoName: (name: string) => void;
+  setErrorType: (error: ErrorType) => void;
+  isAdding: boolean;
+  setErrorClosing: (er: boolean) => void;
+  onUpdate: (todoId: number, done: boolean, title: string) => void;
+};
+
+export const TodoField: React.FC<Props> = ({
+  todos, todoName, onAdd, setNewTodoName,
+  setErrorType, isAdding, setErrorClosing, newToField, onUpdate,
+}) => {
+  const completedAllTodos = todos.every((todo) => todo.completed === true);
+
+  const handleFormSubmit = (event: {
+    preventDefault: () => void;
+  }) => {
+    event.preventDefault();
+
+    if (!todoName.trim()) {
+      setErrorType(ErrorType.Blanc);
+      setErrorClosing(false);
+    }
+
+    if (todoName.trim()) {
+      onAdd(todoName);
+      setErrorType(ErrorType.None);
+    }
+
+    setTimeout(() => newToField.current?.focus(), 500);
+  };
+
+  const handleInput = (event: {
+    target: { value: string; };
+  }) => {
+    setNewTodoName(event.target.value);
+  };
+
+  const handleStatusOnToggle = () => todos.map((todo) => {
+    if (completedAllTodos) {
+      return onUpdate(todo.id, false, todo.title);
+    }
+
+    return onUpdate(todo.id, true, todo.title);
+  });
```

Comment:
```suggestion
  const handleStatusOnToggle = () => (
    todos.map(({ id, title }) => {
      if (completedAllTodos) {
        return onUpdate(id, false, title);
      }

      return onUpdate(id, true, title);
    });
  )
```
Використовуй деструктурування